pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -491,6 +492,10 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 
 	pod := c.pods.getPodByIP(proxyIP)
 	if pod != nil {
+		// for k8s multi cluster, in case there are pods of the same ip across clusters
+		if !strings.HasPrefix(proxy.ID, pod.Name) || !strings.Contains(proxy.ID, pod.Namespace) {
```

Comment:
This doesn't seem sufficient. The real problem is that we can potentially overwrite pods with the same IP, or am I missing something?

Reply:
I am not sure what you mean.  In general, if the pod cidr is same in multi clusters, there may exists multi pods across clusters with same ip, this may lead to `GetProxyServiceInstances` gets the wrong instances for proxies.

Reply:
Right, I understand ... but doesn't this seem like a problem? I suspect it would be racy as to which pod "wins" here ... so we may or may not get the right pod by IP. This seems like a flaw in either the way we're storing or retrieving this information.  The fact that you're returning no instances here seems like a band-aid, but doesn't fix the underlying cause of the problem.  That may be fine if we're purposely limiting the scope of this PR ... but we should be tracking the "real" fix in an issue.  Do we have an issue for this already?

Reply:
Please check the aggregator first, 
>so we may or may not get the right pod by IP.

only one can get the right.
```
func (c *Controller) GetProxyServiceInstances(node *model.Proxy) ([]*model.ServiceInstance, error) {
	out := make([]*model.ServiceInstance, 0)
	var errs error
	// It doesn't make sense for a single proxy to be found in more than one registry.
	// TODO: if otherwise, warning or else what to do about it.
	for _, r := range c.GetRegistries() {
		instances, err := r.GetProxyServiceInstances(node)
		if err != nil {
			errs = multierror.Append(errs, err)
		} else if len(instances) > 0 {
			out = append(out, instances...)
			node.ClusterID = r.ClusterID
			break
		}
	}

	if len(out) > 0 {
		if errs != nil {
			log.Debugf("GetProxyServiceInstances() found match but encountered an error: %v", errs)
		}
		return out, nil
	}

	return out, errs
}

```


>The fact that you're returning no instances here seems like a band-aid, but doesn't fix the underlying cause of the problem. 

I donot know what's meaning here. Why it doesnot fix?

>Do we have an issue for this already?

Absolutely , #14066